Tools/scripts/mhadd

```diff
@@ -111,22 +115,47 @@ def hadd_(is_crab, *args, **kwargs):
         if options.overwrite:
             print colors.yellow('overwriting existing file %s' % new_name)
             rm(new_name)
+        
         else:
             log_fn = new_name + '.haddlog'
+            log_fn_submit = new_name.replace('.root', '') + '.log'
+                        
             if on_eos:
                 log_fn = eos.fusemount(log_fn)
             njobs = hadd_files(working_dir, **kwargs)[0]
+            
             if not exists(log_fn):
                 log_njobs = 1 # assume 1 file copied
-            else:
+                
+            ## helpers to determine which log is present and which is not : submit log for jobs that were submitted to condor
+            ## and haddlog for those that were not
+            s_log = exists(log_fn_submit) and not exists(log_fn)
+            h_log = exists(log_fn) and not exists(log_fn_submit)
+
+            if s_log:
+                 if os.stat(new_name).st_size != 0 :
+                     print colors.boldred('skipping existing file %s which has already been submitted to condor succesfully' % new_name)
+                 #if the root file is created and it is zero, try to run again?      
+                 else:
+                     rm(new_name)
+
+            elif h_log:
                 p = HaddlogParser(log_fn)
                 log_njobs = p.num_sources
-            if log_njobs != njobs:
-                print colors.boldred('skipping existing file %s (for which num sources %i != njobs %i)' % (new_name, log_njobs, njobs))
+
+                if (log_njobs != njobs):
+                    print colors.boldred('skipping existing file %s (for which num sources %i != njobs %i)' % (new_name, log_njobs, njobs))
+                else :
+                    print colors.boldred('skipping existing file %s which has already been hadded succesfully' % new_name)
+                               
             else:
+                #assume that the last case to consider is the one in which files are copied over (which do not produce any log but the .root file does exist)
```

Comment:
For some reason, I find the following error if I have an existing file that I made with `mhadd` locally and try to run via `mhadd -s`:

```
Traceback (most recent call last):
  File "/uscms_data/d3/joeyr/abby_pr/mfv_946p1/bin/slc7_amd64_gcc630/mhadd", line 179, in <module>
    hadd_(is_crab, d, new_dir=x)
  File "/uscms_data/d3/joeyr/abby_pr/mfv_946p1/bin/slc7_amd64_gcc630/mhadd", line 153, in hadd_
    print colors.boldred('skipping existing file %s (for which num sources %i = njobs %i)' % (new_name, log_njobs, njobs))
UnboundLocalError: local variable 'log_njobs' referenced before assignment
```

It looks like `log_njobs` isn't defined when it gets to this line (at least in my case), so mhadd crashes here.